src/k8s/pkg/k8sd/features/coredns/coredns_test.go

```diff
@@ -0,0 +1,181 @@
+package coredns_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/ptr"
+
+	"github.com/canonical/k8s/pkg/client/helm"
+	helmmock "github.com/canonical/k8s/pkg/client/helm/mock"
+	"github.com/canonical/k8s/pkg/client/kubernetes"
+	"github.com/canonical/k8s/pkg/k8sd/features/coredns"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	snapmock "github.com/canonical/k8s/pkg/snap/mock"
+)
+
+func TestDisabled(t *testing.T) {
+	t.Run("HelmApplyFails", func(t *testing.T) {
+		g := NewWithT(t)
+
+		applyErr := errors.New("failed to apply")
+		helmM := &helmmock.Mock{
+			ApplyErr: applyErr,
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+			},
+		}
+		dns := types.DNS{
+			Enabled: ptr.To(false),
+		}
+		kubelet := types.Kubelet{}
+
+		status, str, err := coredns.ApplyDNS(context.Background(), snapM, dns, kubelet, nil)
+
+		g.Expect(err).To(MatchError(ContainSubstring(applyErr.Error())))
+		g.Expect(str).To(BeEmpty())
+		g.Expect(status.Message).To(ContainSubstring(applyErr.Error()))
+		g.Expect(status.Message).To(ContainSubstring("failed to uninstall coredns"))
```

Comment:
we can ignore also ignore the message and not check for that, but it's totally up to you.